snapcraft/models/project.py

```diff
@@ -171,6 +172,32 @@ def _validate_architectures_all_keyword(architectures):
             )
 
 
+def root_packages_transform(yaml_data: dict[str, Any]) -> dict[str, Any]:
+    """Support Root Packages in Snapcraft.
+
+    This allows the user to use "build-packages" and "build-snaps"
+    at the root level of the snapcraft.yaml file.
+    """
+    if "build-packages" not in yaml_data and "build-snaps" not in yaml_data:
+        return yaml_data
+
+    yaml_data = copy.deepcopy(yaml_data)
+    yaml_data.setdefault("parts", {})
+    yaml_data["parts"]["snapcraft/core"] = {"plugin": "nil"}
```

Comment:
How concerned are we with the possibility of someone naming a part `snapcraft/core`? I gather the answer is "not very" since this is the same as the old logic, but it might be prudent to make more liberal use of `setdefault` anyway.

Reply:
we want a predictable name, we need an initial guard to ensure that we only allow / for parts that should be overridden (like extensions).

Not part of this body of work though, please create an issue